query.go

```diff
@@ -892,27 +926,54 @@ func (s *ChainService) GetCFilter(blockHash chainhash.Hash,
 		return nil, err
 	}
 
+	// We will first check if the node supports rest API
+	// and try to get the filter from there
+
 	// With all the necessary items retrieved, we'll launch our concurrent
 	// query to the set of connected peers.
-	log.Debugf("Fetching filters for heights=[%v, %v], stophash=%v",
+	log.Debugf("Fetching filters for heights=[%v, %v], stophash=%v\n",
 		query.startHeight, query.stopHeight, query.stopHash)
 
+	// First attempting to query the rest API and if optimistic batching is
+	// we'll query for blocks
+	// if node does not support the rest API, we'll query using bitcoin's p2p
 	go func() {
 		defer s.mtxCFilter.Unlock()
 		defer close(query.filterChan)
 
-		s.queryPeers(
-			// Send a wire.MsgGetCFilters.
-			query.queryMsg(),
-
-			// Check responses and if we get one that matches, end
-			// the query early.
-			func(_ *ServerPeer, resp wire.Message, quit chan<- struct{}) {
-				s.handleCFiltersResponse(query, resp, quit)
-			},
-			query.options...,
-		)
-
+		if len(s.restPeers) != 0 {
+			quit := make(chan struct{})
+			// We'll need a http client in order to query the host
+			client := &http.Client{Timeout: 10 * time.Second}
```

Comment:
It's perhaps a good idea to use the existing variable QueryTimeout instead of the hard-coded values